pydantic/class_validators.py

```diff
@@ -244,86 +244,46 @@ def make_generic_validator(validator: AnyCallable) -> 'ValidatorCallable':
-        return lambda cls, v, values, field, config: validator(cls, v)
-    elif args == {'values'}:
-        return lambda cls, v, values, field, config: validator(cls, v, values=values)
-    elif args == {'field'}:
-        return lambda cls, v, values, field, config: validator(cls, v, field=field)
-    elif args == {'config'}:
-        return lambda cls, v, values, field, config: validator(cls, v, config=config)
-    elif args == {'values', 'field'}:
-        return lambda cls, v, values, field, config: validator(cls, v, values=values, field=field)
-    elif args == {'values', 'config'}:
-        return lambda cls, v, values, field, config: validator(cls, v, values=values, config=config)
-    elif args == {'field', 'config'}:
-        return lambda cls, v, values, field, config: validator(cls, v, field=field, config=config)
-    else:
-        # args == {'values', 'field', 'config'}
-        return lambda cls, v, values, field, config: validator(cls, v, values=values, field=field, config=config)
-
-
-def _generic_validator_basic(validator: AnyCallable, sig: 'Signature', args: Set[str]) -> 'ValidatorCallable':
-    has_kwargs = False
-    if 'kwargs' in args:
-        has_kwargs = True
-        args -= {'kwargs'}
-
-    if not args.issubset(all_kwargs):
-        raise ConfigError(
-            f'Invalid signature for validator {validator}: {sig}, should be: '
-            f'(value, values, config, field), "values", "config" and "field" are all optional.'
-        )
+    def _fetch_args(args_: Tuple[Any, ...]) -> Dict[str, Any]:
+        return {k: args_[args_order.index(k)] for k in args_order if k in args}
 
-    if has_kwargs:
-        return lambda cls, v, values, field, config: validator(v, values=values, field=field, config=config)
-    elif args == set():
-        return lambda cls, v, values, field, config: validator(v)
-    elif args == {'values'}:
-        return lambda cls, v, values, field, config: validator(v, values=values)
-    elif args == {'field'}:
-        return lambda cls, v, values, field, config: validator(v, field=field)
-    elif args == {'config'}:
-        return lambda cls, v, values, field, config: validator(v, config=config)
-    elif args == {'values', 'field'}:
-        return lambda cls, v, values, field, config: validator(v, values=values, field=field)
-    elif args == {'values', 'config'}:
-        return lambda cls, v, values, field, config: validator(v, values=values, config=config)
-    elif args == {'field', 'config'}:
-        return lambda cls, v, values, field, config: validator(v, field=field, config=config)
-    else:
-        # args == {'values', 'field', 'config'}
-        return lambda cls, v, values, field, config: validator(v, values=values, field=field, config=config)
+    # Actual signature is (cls, value, values, field, config)
+    # cls and value is required, other args is optional
+    return lambda *args_: validator(*args_[skip_first:required_args], **_fetch_args(args_[required_args:]))
```

Comment:
Now that there's ony one left, perhaps it would be "nicer" if this were changed to a "def" instead of passing the lambda around or even just wrap it right away since we already have the validator we need in this function.

```suggestion
    @wraps(validator)
    def validator_wrapper(*args):
        return validator(*args_[skip_first:required_args], **_fetch_args(args_[required_args:]))
        
    return validator_wrapper
```

Reply:
LGTM, it looks much better. I also thought about replacing `lambda` with a regular `def` function.